Add mixed server pre-req

sabuchan07 opened the pull request:
Cherry-pick into:

* [x] Foreman 3.3
* [x] Foreman 3.2
* [ ] Foreman 3.1
* For Foreman 3.0 or older, file a separate PR request

<!---
Thank you for contributing to Foreman documentation. Make sure to read README
for the documentation standards. Set cherry-pick github label to mark this
contribution for cherry picking and check which version do you need with [x].
-->

maximiliankolb commented:
Hi @sabuchan07 Please rebase your PR against master so that you have exactly one commit on your branch (unless you want two separate changes which is perfectly fine!), and have a look at https://cbea.ms/git-commit/#seven-rules. Please ping me when you want me to review this.

sabuchan07 commented:
Hi, @maximiliankolb !  Thank you for the link to the commit message instructions. Very helpful!  Also, forgive my ignorance, but I rebased in my command line, but nothing seems to have changed here.  Is this something I also need to do in the GitHub UI?  Thank you!

maximiliankolb commented:
> 

I assume that you need to fetch and pull from the master branch from Github into your local master branch first, then rebase again.

maximiliankolb commented:
Since you definitely have to rebase to master due to conflicts, you can decide for yourself whether to squash your commits while rebasing or not. I can, when merging, squash your commits and also adjust the commit message.

melcorr commented:
Your local setup doesn't look correct if it is dragging in all this detritus @sabuchan07 
You shouldn't be merging locally as per [15a2956](https://github.com/theforeman/foreman-documentation/pull/1293/commits/15a2956afe18f430d429aa5c58141a858f07c731)
Your commit message also make no sense and we really need them to have relevant message as per https://cbea.ms/git-commit/ - this is included in the setup instructions. 
This looks messy and merging this mess would add mess to our repo.

maximiliankolb commented on guides/common/modules/proc_configuring-repositories.adoc:
```diff
@@ -2,14 +2,16 @@
 ifdef::foreman-el,katello,satellite[]
-* xref:#repositories-rhel-7[{RHEL} 7]
+* xref:#repositories-rhel-7[Red Hat Enterprise Linux 7]
+//* xref:#repositories-rhel-8[Red Hat Enterprise Linux 8]
```
We generally don't add comments like this. Is there anything that readers of the source code should know about el8?

Please either remove it or change it to something like `// _a very imporant/interesting note to maintainers_`.

maximiliankolb commented on guides/common/modules/ref_system-requirements.adoc:
```diff
@@ -41,6 +41,10 @@ endif::[]
 * Full forward and reverse DNS resolution using a fully-qualified domain name
 
+ifdef::satellite[]
+* All {ProjectName} servers must be on the same version.
```
```suggestion
* All {ProjectServer}s must be on the same version.
```

maximiliankolb commented on guides/doc-Upgrading_and_Updating/topics/upgrading_a_disconnected_satellite.adoc:
```diff
@@ -26,6 +26,9 @@ For more information, see https://access.redhat.com/documentation/en-us/red_hat_
 * Back up and remove all Foreman hooks before upgrading.
 Reinstate hooks only after {Project} is known to be working after the upgrade is complete.
+ifdef::satellite[]
+* All {ProjectName} servers must be on the same version.
```
```suggestion
* All {ProjectServer}s must be on the same version.
```

maximiliankolb replied:
This is not resolved yet.

maximiliankolb commented on guides/doc-Upgrading_and_Updating/topics/upgrading_process_overview.adoc:
```diff
@@ -50,6 +50,10 @@ For more information, see {AdministeringDocURL}Backing_Up_Server_and_Proxy_admin
 For more information about changes in the API, see the Red Hat Knowledgebase article https://access.redhat.com/articles/4396911[API Changes Between {Project} Versions] on the Red{nbsp}Hat Customer Portal.
 
+ifndef::satellite[]
+Ensure that all {ProjectName} servers are on the same version.
```
```suggestion
Ensure that all {ProjectServer}s are on the same version.
```

maximiliankolb reviewed (commented):
I made some suggestions to use a different attribute.

maximiliankolb commented on guides/common/modules/proc_configuring-repositories.adoc:
```diff
@@ -2,14 +2,15 @@
 endif::[]
 ifdef::foreman-el,katello,satellite[]
-* xref:#repositories-rhel-7[{RHEL} 7]
+* xref:#repositories-rhel-7[Red Hat Enterprise Linux 7]
```
This is not resolved yet.

maximiliankolb commented on guides/common/modules/proc_configuring-repositories.adoc:
```diff
@@ -2,14 +2,15 @@
 endif::[]
 ifdef::foreman-el,katello,satellite[]
-* xref:#repositories-rhel-7[{RHEL} 7]
+* xref:#repositories-rhel-7[RHEL 7]
```
You are still making a change here that I don't see necessary. `{RHEL}` is an attribute expaning to "Red Hat Enterprise Linux": https://github.com/theforeman/foreman-documentation/blob/master/guides/common/attributes-base.adoc#L90

sabuchan07 replied:
Sorry about that...that one was just me not thinking.

sabuchan07 commented:
@maximiliankolb Is this one ready to merge?

maximiliankolb reviewed (commented):
Made some more suggestions. Please apply those, close the threads, and rebase to master. To craft your commit message, have a look at https://cbea.ms/git-commit/#seven-rules

maximiliankolb commented on guides/common/modules/proc_configuring-repositories.adoc:
```diff
@@ -2,15 +2,21 @@
+=======
+* xref:#repositories-rhel-7[Red Hat Enterprise Linux 7]
+//* xref:#repositories-rhel-8[Red Hat Enterprise Linux 8]
+>>>>>>> 7b850960 (Added info about not supporting mixed version servers)
```
This does not look correct. It looks like you had a merge conflict and did not correctly resolve it. `<<<<`, `====`, and `>>>>>` are markers from git which must not be commited.

maximiliankolb commented on guides/common/modules/ref_system-requirements.adoc:
```diff
@@ -41,9 +41,15 @@ endif::[]
 {ProjectServer} and {SmartProxyServer} do not support shortnames in the hostnames.
 When using custom certificates, the Common Name (CN) of the custom certificate must be a fully qualified domain name (FQDN) instead of a shortname.
 This does not apply to the clients of a {Project}.
+=======
```
same as above.

maximiliankolb commented on guides/common/modules/proc_configuring-repositories.adoc:
```diff
@@ -2,15 +2,21 @@
 * xref:#repositories-rhel-7[{RHEL} 7]
+=======
+* xref:#repositories-rhel-7[Red Hat Enterprise Linux 7]
+//* xref:#repositories-rhel-8[Red Hat Enterprise Linux 8]
```
Is this meant to be a comment? We generally don't add comments like this. Either we include it, or not.

maximiliankolb reviewed (changes requested):
Almost there, just some git markers left in there.

maximiliankolb commented:
@sabuchan07 There are currently some merge conflicts. Please rebase to master and reword you commit message.

sabuchan07 commented:
@maximiliankolb Fixed the conflicts and rebased but, as you can clearly see, it's now added 250+ commits.  Just want you to know I'm aware of what happened, and I'll fix by squashing commits now.

ekohl commented:
@sabuchan07 perhaps you rebased on the wrong branch? Forgot to fetch first? I'd try
```
# See which remote is theforeman/foreman-documentation
git remote -v
# Assuming it's orign
git fetch origin
git rebase origin/master
```

Then see which commits are left:
```
git log origin/master..HEAD
```

maximiliankolb commented on guides/common/modules/con_about-running-jobs-on-hosts.html:
```diff
@@ -0,0 +1,496 @@
+<!DOCTYPE html>
```
you'll want to remove this file.

sabuchan07 commented:
@ekohl Thanks for your suggestion!  I did all of that and, unfortunately, it's still listing a ton of commits.  Going to go about trying to remove them now.

ekohl commented:
Ping me downstream if you want to go over this using screen sharing.

sabuchan07 commented:
Hi, @maximiliankolb !  Is this one ready to merge?  Thanks!

maximiliankolb commented on guides/doc-Upgrading_and_Updating/topics/upgrading_overview.adoc:
```diff
@@ -38,6 +38,10 @@ For more information, see {AdministeringDocURL}Backing_Up_Server_and_Proxy_admin
 For more information about changes in the API, see the Red Hat Knowledgebase article https://access.redhat.com/articles/4396911[API Changes Between {Project} Versions] on the Red{nbsp}Hat Customer Portal.
 
+ifndef::satellite[]
+Ensure that all {ProjectName} servers are on the same version.
```
```suggestion
Ensure that all {ProjectServer}s are on the same version.
```

See https://github.com/theforeman/foreman-documentation/blob/master/guides/common/attributes-base.adoc#L82

maximiliankolb replied:
same as above.

maximiliankolb reviewed (changes requested):
This needs some additional work.

sabuchan07 commented:
@maximiliankolb Apologies for missing those.  It should be all set now.

maximiliankolb reviewed (changes requested):
Please remove "guides/common/modules/con_about-running-jobs-on-hosts.html" from your branch.

please also rebase and squash your commit and reword your commit message.

maximiliankolb commented:
Unfortunately your PR seems still broken; I can see a lot more changes in the diff than expected.

sabuchan07 commented:
@maximiliankolb This is all set now.

maximiliankolb commented:
I think the change from "ProjectName servers" has been reverted; it should be "{ProjectServer}s".

sabuchan07 commented:
@maximiliankolb Fixed!

maximiliankolb commented on guides/doc-Upgrading_and_Updating/topics/upgrading_overview.adoc:
```diff
@@ -38,6 +38,10 @@ For more information, see {AdministeringDocURL}Backing_Up_Server_and_Proxy_admin
 * Plan for updating any scripts you use that contain {Project} API commands because some API commands differ between versions of {Project}.
 For more information about changes in the API, see the Red Hat Knowledgebase article https://access.redhat.com/articles/4396911[API Changes Between {Project} Versions] on the Red{nbsp}Hat Customer Portal.
 
+ifndef::satellite[]
```
```suggestion
ifdef::satellite[]
```

maximiliankolb commented on guides/doc-Upgrading_and_Updating/topics/upgrading_overview.adoc:
```diff
@@ -79,7 +79,7 @@ ifdef::katello,orcharhino,satellite[]
 For more information, see xref:upgrading_content_hosts[].
 endif::[]
-ifndef::foreman-deb,orcharhino[]
+ifdef::foreman-deb,orcharhino[]
```
please revert this change.

maximiliankolb commented on guides/doc-Upgrading_and_Updating/topics/upgrading_overview.adoc:
```diff
@@ -57,7 +57,7 @@ You can upgrade to {ProjectName} {ProjectVersion} from {ProjectName} {ProjectVer
 endif::[]
 
-ifndef::satellite[]
+ifdef::satellite[]
```
please revert this change.

sabuchan07 commented:
@maximiliankolb Additional changes made

maximiliankolb reviewed (approved):
LGTM; I'll squash myself.

maximiliankolb commented:
Merged to master and cherry-picked: ✅ 
   ca5e7a591..c7382bdc3  3.4 -> 3.4
   ea5b39c14..8962ce391  3.3 -> 3.3
   55fade215..ace2f6827  3.2 -> 3.2
